test/verifying_paymaster.test.ts

```diff
@@ -37,19 +41,30 @@ describe('EntryPoint with VerifyingPaymaster', function () {
     ({ proxy: account } = await createAccount(ethersSigner, accountOwner.address, entryPoint.address))
   })
 
+  describe('#parsePaymasterAndData', () => {
+    it('should parse data properly', async () => {
+      const paymasterAndData = hexConcat([paymaster.address, defaultAbiCoder.encode(['uint64', 'uint64'], [MOCK_VALID_UNTIL, MOCK_VALID_AFTER]), MOCK_SIG])
+      console.log(paymasterAndData)
+      const res = await paymaster.parsePaymasterAndData(paymasterAndData)
+      expect(res.validUntil).to.be.equal(ethers.BigNumber.from(MOCK_VALID_UNTIL))
+      expect(res.validAfter).to.be.equal(ethers.BigNumber.from(MOCK_VALID_AFTER))
+      expect(res.signature).equal(MOCK_SIG)
+    })
+  })
+
   describe('#validatePaymasterUserOp', () => {
     it('should reject on no signature', async () => {
       const userOp = await fillAndSign({
         sender: account.address,
-        paymasterAndData: hexConcat([paymaster.address, '0x1234'])
+        paymasterAndData: hexConcat([paymaster.address, defaultAbiCoder.encode(['uint64', 'uint64'], [MOCK_VALID_UNTIL, MOCK_VALID_AFTER]), '0x1234'])
```

Comment:
you keep the old uint64, which is confusing.
it is uint48 now.

Reply:
good catch